sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -325,9 +333,21 @@ void guessLocalWorkSize(_pi_device *device, size_t *threadsPerBlock,
       std::min(maxThreadsPerBlock[0],
                std::min(global_work_size[0], static_cast<size_t>(gridDim[0])));
 
+  // Query the warp-size to use as preferred work-group size multiple.
+  int warpSize{0};
+  PI_CHECK_ERROR(cuDeviceGetAttribute(&warpSize, CU_DEVICE_ATTRIBUTE_WARP_SIZE,
+                                      device->get()));
+  const size_t preferredMultiple = warpSize;
+
   // Find a local work group size that is a divisor of the global
   // work group size to produce uniform work groups.
-  while (0u != (global_work_size[0] % threadsPerBlock[0])) {
+  // Additionally, the best produced local size:
+  // - has to be multiple of the Cuda warp size
+  // - or any power of 2, if still not evenly distributed.
```

Comment:
This comment seems to imply that just one of those is sufficient, but the while loop below goes on until both conditions are satisfied (or, for the first one, that the local WG size is less than the warp size). Which one is correct?

Reply:
I will update this. The meaning of the comment is the correct one here.
We will get a multiple of warp size if the global work size allows anyways, but if not then it is best to go with power of 2, if possible, for better utilisation of processing elements. For example, in the issue linked above a wg-size of 8 with more groups is better than wg-size of 344 (highest possible with uniformly distributed groups).
So, actually, the warp size check is redundant here, because we still want loop until a power of two value anyways. I'll rid of the condition and queries wrt warp-size here.

I think I got wrapped up in it initially, so thanks for questioning this! @sergey-semenov 
For extra testing, I ran BabelStream with SYCL2020 backend and have not seen any regressions.